pkg/apis/extensions/v1beta1/defaults.go

```diff
@@ -57,6 +57,17 @@ func SetDefaults_DaemonSet(obj *extensionsv1beta1.DaemonSet) {
 			updateStrategy.RollingUpdate.MaxUnavailable = &maxUnavailable
 		}
 	}
+	if updateStrategy.Type == extensionsv1beta1.SurgingRollingUpdateDaemonSetStrategyType {
+		if updateStrategy.SurgingRollingUpdate == nil {
+			rollingUpdate := extensionsv1beta1.SurgingRollingUpdateDaemonSet{}
+			updateStrategy.SurgingRollingUpdate = &rollingUpdate
```

Comment:
Just curious, why set to the variable `rollingUpdate` which is only used in this scope?